### PR TITLE
Fix to TL1 related to interpretation of "Julian day"

### PR DESCRIPTION
**Pull request by dblodgett-usgs**
See new "Note" in documentation of calc_timingLow.R and change from using day of water year to using day of gregorian calendar year.  

More information in #145.

### CODE REVIEW

File: R/calc_timingLow.R [hunk 1]
```diff
@@ -40,21 +42,24 @@ calc_timingLow <- function(x,yearType = "water",digits=3,pref="mean",floodThresh
...
         flowSum_year$mdata <- sin(flowSum_year$minFlowJulDay*2*pi/365.25)
         
+        # Average accross all years.
         xbar <- mean(flowSum_year$np)
         ybar <- mean(flowSum_year$mdata)
-        if (xbar>0) {
+        if (xbar>0) { # if x component is greater than 0, just return the angle.
                 tl1_temp <- atan(ybar/xbar)*180/pi
-        } else if (xbar<0) {
+        } else if (xbar<0) { # if x component is less than 0, arctan returns less than 180 but days should be
```
Comment (wdwatkins) [inline R/calc_timingLow.R]: Does this comment need to be finished?

File: tests/testthat/tests_calc_timingLow.R [hunk 2]
```diff
@@ -5,17 +5,26 @@ test_that("null flood threshold", {
         x<-sampleData[c("date","discharge")]
         
         calc_timingLowOutTest <- calc_timingLow(x=x,yearType="water")
-        calc_timingLowOut <- readRDS("data/tests_calc_timingLow_null.rds")
-       
-        expect_equal(calc_timingLowOutTest,calc_timingLowOut)
+
+        expect_equal_to_reference(calc_timingLowOutTest,"data/tests_calc_timingLow_null.rds")
```
Comment (wdwatkins) [inline tests/testthat/tests_calc_timingLow.R]: 👍

**Comment by dblodgett-usgs** [comment]
You mean in the documentation? I'm hesitant to do that since the documentation is pulled directly from the canonical reference. We could... but that would be a first.

File: R/calc_timingLow.R (hunk 1 above)
Comment (dblodgett-usgs) [inline R/calc_timingLow.R]: Yes... derp.

**Comment by wdwatkins** [comment]
Ah ok.  Day of year/gregorian day is the more technically correct term here though right?

**Comment by dblodgett-usgs** [comment]
Yeah... probably. In the note that I added:

`Note: In these definitions, “Julian date” should be interpreted as the count of days starting with 1 on January first of a given year, ending at 365 or 366 on December 31st.`

I avoided using the gregorian term and am just trying to be unambiguous about it in plain terms.

File: R/calc_timingLow.R [hunk 3]
```diff
@@ -40,30 +42,38 @@ calc_timingLow <- function(x,yearType = "water",digits=3,pref="mean",floodThresh
...
+        # Average accross all years.
         xbar <- mean(flowSum_year$np)
         ybar <- mean(flowSum_year$mdata)
-        if (xbar>0) {
+        if (xbar>0) { # if x component is greater than 0, just return the angle.
                 tl1_temp <- atan(ybar/xbar)*180/pi
-        } else if (xbar<0) {
+        } else if (xbar<0) { # if x component is less than 0 (angles pi/2 to 3pi/2), 
+                             # arctan returns -90 -> 0 -> 90 but we should return
+                             # 90 -> 180 -> 270 so add 180 to output.
```
Comment (dblodgett-usgs) [inline R/calc_timingLow.R]: For the record, I had to look at a unit circle and do this:
```
# just passed 90 degrees
> sin(1.1*pi/2)
[1] 0.9876883
> cos(1.1*pi/2)
[1] -0.1564345
> atan(0.98/-0.15)*180/pi
[1] -81.29778
# need to add 180 to get to near 90.

# Just before 270 degrees
> sin(2.9*pi/2)
[1] -0.9876883
> cos(2.9*pi/2)
[1] -0.1564345
> atan(-0.98/-0.15)*180/pi
[1] 81.29778
# need to add 180 to get near 270.

# Approaching 180 degrees
> sin(1.9*pi/2)
[1] 0.1564345
> cos(1.9*pi/2)
[1] -0.9876883
> atan(0.15/-0.98)*180/pi
[1] -8.702224
# need to add 180 to get near 180
```

File: R/calc_timingLow.R [hunk 4]
```diff
@@ -40,30 +42,38 @@ calc_timingLow <- function(x,yearType = "water",digits=3,pref="mean",floodThresh
...
                 tl1_temp <- (atan(ybar/xbar)*180/pi)+180
-        } else if (xbar==0 && ybar>0) {
+        } else if (xbar==0 && ybar>0) { # discontinuity
                 tl1_temp <- 90
-        } else if (xbar==0 && ybar<0) {
+        } else if (xbar==0 && ybar<0) { # discontinuity
                 tl1_temp <- 270
         }
         
+        # if tl1_temp is negative from above (for 270 or 3pi/2 to 360 or 2pi degrees) just add 360.
```
Comment (dblodgett-usgs) [inline R/calc_timingLow.R]: This code could probably be refactored to clean up all this trig if/else mess.

**Pull request merged**